src/models/imports.ts

```diff
@@ -0,0 +1,126 @@
+import Joi from 'joi';
+import { PinGroup } from './pin-group';
+import { EnvironmentReportType } from './environment-report-type';
+
+type Template = {
+  pinGroupHashIds: Array<PinGroup['hashId']>,
+  reportTypeHashIds: Array<EnvironmentReportType['hashId']>,
+}
+
+type BaseImport = {
+  name: string
+  hashId: string
+  createdAt: Date
+  updatedAt: Date
+  deletedAt?: Date | null
+}
+
+/**
+  * The import is being processed. There is no link available at that time.
+  */
+type ProcessingImport = BaseImport & {
+  state: 'processing'
+}
+
+/**
+  * The processing of the import failed early.
+  * There could be several reasons for this such as invalid file format
+  */
+type InvalidImport = BaseImport & {
+  state: 'invalid'
+  errors: Array<string>
+}
+
+/**
+  * The processing of the import failed on one or more rows.
+  * There is a link available for the provided XLSX file where failed rows
+  * have an explanation of the error.
+  */
+type ErroredImport = BaseImport & {
+  state: 'errored'
+  link: string
+  processedAt: Date
+}
+
+/**
+  * The processing of the import is successful for all rows.
+  * There is a link available for the provided XLSX file.
+  */
+type SuccessfulImport = BaseImport & {
+  state: 'success'
+  link: string
+  processedAt: Date
+}
+
+type Import = ProcessingImport
+  | InvalidImport
+  | ErroredImport
+  | SuccessfulImport
+
+const templateSchema = Joi.object().keys({
+  pinGroupHashIds: Joi.array().items(Joi.string().required().example('xd2rd4')).required(),
+  reportTypeHashIds: Joi.array().items(Joi.string().required().example('xd2rd4')).required(),
+});
+
+const baseImportSchema = Joi.object().keys({
+  hashId: Joi.string().required().example('xd2rd4'),
```

Comment:
Convention currently is to make up a new hashId for a new model and use that consistently.

Reply:
Does the convention defines a way to "pick" the hashId ?

Does it increment ? (if yes, how can I find the latest number)
or 
Should I just pick a random hashId and look if it's used somewhere ?

Reply:
The latter ;)